lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -98,3 +108,185 @@ def _validate_and_link(program_uuid_string, linkage_text):
         ]
         errors = [message for message in link_errors.values()]
         return successes, errors
+
+
+class ProgramEnrollmentsInspectorView(View):
+    """
+    The view to search and display the program enrollments
+    information of a learner.
+    """
+    exclude_from_schema = True
+    CONSOLE_TEMPLATE_PATH = 'support/program_enrollments_inspector.html'
+
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        # We need to get some SSO organizations
+        return render_to_response(
+            self.CONSOLE_TEMPLATE_PATH,
+            {
+                'successes': [],
+                'errors': [],
+                'learner_program_enrollments': '',
+                'org_keys': self._get_org_keys_with_IdP_provider(),
+            }
+        )
+
+    @method_decorator(require_support_permission)
+    def post(self, request):
+        """
+        Find the learner and the corresponding ProgramEnrollment info
+        based on the posted organization short name and learner's external user key
```

Comment:
This reads like you're not actually changing anything with this HTTP POST. Do I have that right?

Reply:
Could we make the form a get instead and wrap these two methods up with a conditional based on query params?

Reply:
Feedback taken. @kdmccormick helped me find that we can put GET on Form methods attributes. It works wonderfully. Thank you